internal/plan/plan.go

```diff
@@ -184,7 +184,7 @@ func (s *Service) Merge(other *Service) {
 	if other.BackoffLimit.IsSet {
 		s.BackoffLimit = other.BackoffLimit
 	}
-	s.LogTargets = appendUnique(s.LogTargets, other.LogTargets...)
+	s.LogTargets = MergeLogTargets(s.LogTargets, other.LogTargets)
```

Comment:
Could this should be `s.LogTargets.Merge(other.LogTargets)`, rather than a stand-alone function, for consistency with the other `Merge` methods? `Merge` is a method on the pointer receiver, so it can modify the object as needed.

Reply:
Originally I had `s.LogTargets` as a pointer, but kept running into nil pointer dereference issues. Easy fix was to make it a value field instead. I could change it back to a pointer and just be more vigilant about checking for nil before every operation.

Reply:
I don't think we need the `LogTargets` field to be a pointer to do this ([example](https://go.dev/play/p/JKGF0xTslLv)). I'm just referring to `Merge` being a method and having a similar signature to the other `Merge` methods.